### PR TITLE
binary: complete encoder

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by mathetake** [comment]
I need this to accelerate writing and debugging the random module generator for testing

**Comment by mathetake** [comment]
ok 100% now binary.EncodeModule  produces exactly the same binary as the original source module custom sections 🕺

**Comment by mathetake** [comment]
for all the binaries in spectests

File: internal/wasm/binary/const_expr.go [hunk 1]
```diff
@@ -48,10 +48,17 @@ func decodeConstantExpression(r *bytes.Reader) (*wasm.ConstantExpression, error)
 		return nil, fmt.Errorf("constant expression has been not terminated")
 	}
 
-	data := make([]byte, remainingBeforeData-int64(r.Len()))
+	data := make([]byte, remainingBeforeData-int64(r.Len())-1)
```
Comment (mathetake) [inline internal/wasm/binary/const_expr.go:51]: data doesn't need OpcodeEnd :D

**Comment by mathetake** [comment]
seems like I need to fix more 😄

**Comment by mathetake** [comment]
ok fixed 🤞

File: internal/wasm/binary/const_expr.go (hunk 1 above)
Comment (codefromthecrypt) [reply internal/wasm/binary/const_expr.go:51]: thanks!

**Review by codefromthecrypt — APPROVED** [review]
excellent thanks!

**Pull request merged**